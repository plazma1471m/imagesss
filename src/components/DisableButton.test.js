import React from 'react'
import DisableButton from './DisableButton'
import renderer from 'react-test-renderer'

it('renders disable button correctly', () => {
  const tree = renderer
    .create(<DisableButton disabledHosts={[]} currentUrl="https://google.com/foo.png" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders enable button correctly', () => {
  const tree = renderer
    .create(
      <DisableButton disabledHosts={['google.com']} currentUrl="https://google.com/foo.png" />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('does not render on non-URL pages', () => {
  const tree = renderer
    .create(<DisableButton disabledHosts={[]} currentUrl="file:///foo.png" />)
    .toJSON()
  expect(tree).toBeNull()
})
