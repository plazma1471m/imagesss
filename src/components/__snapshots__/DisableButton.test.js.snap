// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders disable button correctly 1`] = `
<div
  className="ui attached segment"
>
  <button
    className="ui basic fluid positive button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
  >
    Disable on this site
  </button>
</div>
`;

exports[`renders enable button correctly 1`] = `
<div
  className="ui attached segment"
>
  <button
    className="ui basic fluid negative button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
  >
    Enable on this site
  </button>
</div>
`;
